### PR TITLE
 [vulkan] Make vkGetInstanceProcAddr a required config param.

### PR DESCRIPTION
**Pull request by dj2**
This CL changes the VulkanConfig to require the vkGetInstanceProcAddr
pointer. This will be used internally to lookup all the used vulkan
methods.

### CODE REVIEW

File: include/amber/amber_vulkan.h [hunk 1]
```diff
@@ -43,6 +43,9 @@ struct VulkanEngineConfig : public EngineConfig {
 
   /// The VkQueue to use.
   VkQueue queue;
+
+  /// REQUIRED. The Vulkan instance procedue loader.
```
Comment (dneto0) [inline include/amber/amber_vulkan.h]: "procedure"

File: samples/config_helper_vulkan.cc [hunk 2]
```diff
@@ -18,6 +18,7 @@
 #include <cassert>
 #include <cstring>
 #include <set>
+#include <vulkan/vulkan.h>
```
Comment (dneto0) [inline samples/config_helper_vulkan.cc]: Hmmm. I normally use " quotes for things not shipped with the compiler.   But others have argued against me.
Coder's choice here.

**Comment by dneto0** [comment]
Recommend fixing the typo, and making sure formatting is acceptable (however headers are arranged).
But otherwise LGTM.

File: include/amber/amber_vulkan.h (hunk 1 above)
Comment (dj2) [reply include/amber/amber_vulkan.h]: done. I'd actually fixed it in the next CL in the chain already, heh.

File: samples/config_helper_vulkan.cc (hunk 2 above)
Comment (dj2) [reply samples/config_helper_vulkan.cc]: I usually go with ""'s for things in the repo (so in include/ src/ or third_party/) and <> otherwise

**Pull request merged**